src/components/accordion/accordion.jsx

```diff
@@ -0,0 +1,146 @@
+import React, { useState } from 'react';
+import { Plus, Minus, ChevronDown } from 'lucide-react';
+import { cn } from '@/utilities/functions';
+
+const Accordion = ( props ) => {
+	const {
+		type = 'simple',
+		defaultValue,
+		disabled = false,
+		children,
+		className,
+	} = props;
+	const [ activeItem, setActiveItem ] = useState( defaultValue );
+
+	const handleToggle = ( value ) => {
+		setActiveItem( ( prev ) => ( prev === value ? null : value ) );
+	};
+
+	const typeClasses = type === 'boxed' ? 'space-y-3' : '';
+
+	return (
+		<div className={ cn( typeClasses, className ) }>
+			{ React.Children.map( children, ( child ) =>
+				React.cloneElement( child, {
+					isActive: child.props.value === activeItem,
+					onToggle: () => handleToggle( child.props.value ),
+					type,
+					disabled: disabled || child.props.disabled,
+				} )
+			) }
+		</div>
+	);
+};
+
+Accordion.displayName = 'Accordion';
+
+const AccordionItem = ( props ) => {
+	const {
+		isActive,
+		onToggle,
+		type,
+		value,
+		disabled = false,
+		children,
+		className,
+	} = props;
+
+	let typeClasses = 'border-0';
+	if ( type === 'separator' ) {
+		typeClasses = 'border-0 border-b border-solid border-border-subtle';
+	} else if ( type === 'boxed' ) {
+		typeClasses = 'border border-solid border-border-subtle rounded-md';
+	}
+
```

Comment:
can we use something like below

const classtomap = {
  separator: 'border-0 separator classes',
  boxed: 'boxed classe',
  default: default classes
};

const typeClasses = classtomap[type] || classtomap.default;